src/BenchmarkDotNet/Environments/Runtimes/CoreRuntime.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Portability;
+
+namespace BenchmarkDotNet.Environments
+{
+    public class CoreRuntime : Runtime
+    {
+        public static readonly CoreRuntime Core20 = new CoreRuntime(TargetFrameworkMoniker.NetCoreApp20, "netcoreapp2.0", ".NET Core 2.0");
+        public static readonly CoreRuntime Core21 = new CoreRuntime(TargetFrameworkMoniker.NetCoreApp21, "netcoreapp2.1", ".NET Core 2.1");
+        public static readonly CoreRuntime Core22 = new CoreRuntime(TargetFrameworkMoniker.NetCoreApp22, "netcoreapp2.2", ".NET Core 2.2");
+        public static readonly CoreRuntime Core30 = new CoreRuntime(TargetFrameworkMoniker.NetCoreApp30, "netcoreapp3.0", ".NET Core 3.0");
+        public static readonly CoreRuntime Core31 = new CoreRuntime(TargetFrameworkMoniker.NetCoreApp31, "netcoreapp3.1", ".NET Core 3.1");
+        public static readonly CoreRuntime Core50 = new CoreRuntime(TargetFrameworkMoniker.NetCoreApp50, "netcoreapp5.0", ".NET Core 5.0");
+
+        private CoreRuntime(TargetFrameworkMoniker targetFrameworkMoniker, string msBuildMoniker, string displayName)
+            : base(targetFrameworkMoniker, msBuildMoniker, displayName)
+        {
+        }
+
+        /// <summary>
+        /// use this method if you want to target .NET Core version not supported by current version of BenchmarkDotNet. Example: .NET Core 10
+        /// </summary>
+        /// <param name="msBuildMoniker">msbuild moniker, example: netcoreapp10.0</param>
+        /// <param name="displayName">display name used by BDN to print the results</param>
+        /// <returns>new runtime information</returns>
+        public static CoreRuntime CreateForNewVersion(string msBuildMoniker, string displayName)
+        {
+            if (string.IsNullOrEmpty(msBuildMoniker)) throw new ArgumentNullException(nameof(msBuildMoniker));
+            if (string.IsNullOrEmpty(displayName)) throw new ArgumentNullException(nameof(displayName));
+
+            return new CoreRuntime(TargetFrameworkMoniker.NotRecognized, msBuildMoniker, displayName);
+        }
+
+        internal static CoreRuntime GetCurrentVersion()
+        {
+            if (!RuntimeInformation.IsNetCore)
+            {
+                throw new NotSupportedException("It's impossible to reliably detect the version of .NET Core if the process is not a .NET Core process!");
+            }
+
+            string netCoreAppVersion = RuntimeInformation.GetNetCoreVersion();
+            if (!string.IsNullOrEmpty(netCoreAppVersion) && Version.TryParse(netCoreAppVersion, out _))
```

Comment:
This doesn't work for current prerelease versions (eg. `5.0.0-alpha1.19426.15`). It should have the same `.TakeWhile(c => char.IsDigit(c) || c == '.')` treatment as the version extracted later in `FromProductVersion`.

Reply:
@filipnavara great catch! I've send a separate PR with the fix https://github.com/dotnet/BenchmarkDotNet/pull/1233